File: Lab1/Bloggo.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Bloggo {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String texto;

        while ((texto = br.readLine()) != null) {
            StringBuilder saidaTexto = new StringBuilder();
            boolean italico = false;
            boolean negrito = false;

            for (char c : texto.toCharArray()) {
                if (c == '_') {
                    if (italico) {
                        saidaTexto.append("</i>");
                    } else {
                        saidaTexto.append("<i>");
                    }
                    italico = !italico;
                } else if (c == '*') {
                    if (negrito) {
                        saidaTexto.append("</b>");
                    } else {
                        saidaTexto.append("<b>");
                    }
                    negrito = !negrito;
                } else {
                    saidaTexto.append(c);
                }
            }

            System.out.println(saidaTexto.toString());
        }
    }
}

File: Lab1/Bloggo.cpp
#include <iostream>
#include <string>

int main() {
    std::string texto;
    
    while (std::getline(std::cin, texto)) {
        std::string saidaTexto;
        bool italico = false;
        bool negrito = false;

        for (char c : texto) {
            if (c == '_') {
                if (italico) {
                    saidaTexto += "</i>";
                } else {
                    saidaTexto += "<i>";
                }
                italico = !italico;
            } else if (c == '*') {
                if (negrito) {
                    saidaTexto += "</b>";
                } else {
                    saidaTexto += "<b>";
                }
                negrito = !negrito;
            } else {
                saidaTexto += c;
            }
        }

        std::cout << saidaTexto << std::endl;
    }

    return 0;
}
